File: server/Rekindle.Core/Models/Media.cs
using System.Text.Json.Serialization;

namespace Rekindle.Core.Models;

public class Media
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string LibraryId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Series { get; set; }
    public int? Volume { get; set; }

    // Absolute server filesystem paths — internal only. [JsonIgnore] keeps the
    // server's directory layout out of API responses that any read user can see.
    [JsonIgnore] public string FilePath { get; set; } = string.Empty;

    public string Format { get; set; } = string.Empty;
    public int? PageCount { get; set; }

    [JsonIgnore] public string? CoverCachePath { get; set; }

    public string MediaType { get; set; } = "archive"; // "archive" | "folder"
    public string RelativePath { get; set; } = string.Empty;
    public string? ParentId { get; set; }
    public DateTime AddedAt { get; set; } = DateTime.UtcNow;
}

File: desktop/lib/core/models/media.dart
class Media {
  final String id;
  final String libraryId;
  final String title;
  final String? series;
  final int? volume;
  final String format;
  final int? pageCount;
  final String? coverCachePath;
  final String mediaType; // "archive" | "folder"
  final String relativePath; // path relative to library root, e.g. "Absolute Batman/Chapter1.cbz"
  final String? parentId;

  const Media({
    required this.id,
    required this.libraryId,
    required this.title,
    this.series,
    this.volume,
    required this.format,
    this.pageCount,
    this.coverCachePath,
    this.mediaType = 'archive',
    this.relativePath = '',
    this.parentId,
  });

  factory Media.fromJson(Map<String, dynamic> j) => Media(
        id: j['id'] as String,
        libraryId: j['libraryId'] as String,
        title: j['title'] as String,
        series: j['series'] as String?,
        volume: j['volume'] as int?,
        format: j['format'] as String,
        pageCount: j['pageCount'] as int?,
        coverCachePath: j['coverCachePath'] as String?,
        mediaType: j['mediaType'] as String? ?? 'archive',
        relativePath: j['relativePath'] as String? ?? '',
        parentId: j['parentId'] as String?,
      );

  String get displayTitle {
    if (series != null && volume != null) return '$series #$volume';
    return title;
  }

  bool get isFolder => mediaType == 'folder';
  bool get isImageBased => ['cbz', 'cbr', 'pdf'].contains(format);
}
